### PR TITLE
[catch2] Guard build type for Catch2

### PR DESCRIPTION
**Pull request by westfly**
**Describe the pull request**
Avoid Catch2 build failed On only Release Mode
- #### What does your PR fix?  
  Fixes #... https://github.com/microsoft/vcpkg/issues/19778

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
`All`, `No`

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

### CODE REVIEW

**Review by NancyLi1013 — COMMENTED** [review]
Could you please also update the version in ports/catch2.json?

Add `"port-version" : 1,` below `version-semver` field.

**Comment by westfly** [comment]
> port-version
I'm freshman on port.
when add port version, the status changed

image

I think it's right

**Comment by NancyLi1013** [comment]
The version is right. Could you please format the manifest?

Run `.\vcpkg format-manifest ./ports/catch2/vcpkg.json` and then submit the changes.

You also need to update version files via `.\vcpkg x-add-version catch2`.

**Comment by westfly** [comment]
> The version is right. Could you please format the manifest?
> 
> Run `.\vcpkg format-manifest ./ports/catch2/vcpkg.json` and then submit the changes.
> 
> You also need to update version files via `.\vcpkg x-add-version catch2`.

Thx for your guidance  with great patient.

**Review by NancyLi1013 — APPROVED** [review]

File: versions/c-/catch2.json [hunk 1]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "900833643e0eb468d957b4c46f46ba6647e283ff",
+      "git-tree": "4df380019e5c3a0e450798db43d9ce70906b7aad",
+      "version-semver": "2.13.7",
+      "port-version": 1
+    },
+    {
+      "git-tree": "eea1021d4b3c4e41919b5e96a4372604e5837979",
```
Comment (NancyLi1013) [inline versions/c-/catch2.json:9]:
```suggestion
      "git-tree": "900833643e0eb468d957b4c46f46ba6647e283ff",
```
This seems not right. It should not change. The new item should be added to the top of original ones. Could you please revert this change and use` ./vcpkg x-add-version --overwrite-version catch2` to update this?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
I'm sorry. There is still one point that needs to update. Please see my last review suggestion @westfly.

File: versions/c-/catch2.json (hunk 1 above)
Comment (westfly) [reply versions/c-/catch2.json:9]: I run command one step by step as below, am i right ?

1. change ports/catch2/portfile.cmake
2. add port-version on ports/catch2/vcpkg.json
3. vcpkg format-manifest ports/*/vcpkg.json
4. vcpkg x-add-version --overwrite-version catch2
5. git commit -m "update baseline & version" ports/ versions
6. git push

Comment (NancyLi1013) [reply versions/c-/catch2.json:9]: You should commit the changes(in ports/catch2/portfile.cmake and ports/catch2/vcpkg.json) first, then run `vcpkg x-add-version --overwrite-version catch2` and commit the changes about versions. Last, git push the commits. Or you can push the commit after you commit the one.

Comment (westfly) [reply versions/c-/catch2.json]: I redo command by adding "git commit" between step 3 and 4 , and push by force.  thx for your reply

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM, thanks for your PR. @westfly
